File: frontend/src/objects/order/core/OrderSummaryReviewContent.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  DescriptionText,
  IsoDateTime,
  NoteText,
} from '@/objects/core/SharedObjects'

export type OrderSummaryReviewContent = {
  reviewComment?: DescriptionText
  reviewExtraNote?: NoteText
  storeReviewComment?: DescriptionText
  storeReviewExtraNote?: NoteText
  storeMerchantReply?: NoteText
  storeMerchantReplyAt?: IsoDateTime
  riderReviewComment?: DescriptionText
  riderReviewExtraNote?: NoteText
}

File: backend/src/main/scala/domain/order/OrderSummaryReviewContent.scala
package domain.order

import domain.shared.*

final case class OrderSummaryReviewContent(
    reviewComment: Option[ReasonText],
    reviewExtraNote: Option[NoteText],
    storeReviewComment: Option[ReasonText],
    storeReviewExtraNote: Option[NoteText],
    storeMerchantReply: Option[NoteText],
    storeMerchantReplyAt: Option[IsoDateTime],
    riderReviewComment: Option[ReasonText],
    riderReviewExtraNote: Option[NoteText],
)
